pinot-core/src/main/java/com/linkedin/pinot/core/realtime/impl/kafka/SimpleConsumerWrapper.java

```diff
@@ -102,30 +100,18 @@ public TransientConsumerException(Errors error) {
     }
   }
 
-  private SimpleConsumerWrapper(KafkaSimpleConsumerFactory simpleConsumerFactory, String bootstrapNodes,
-      String clientId, long connectTimeoutMillis) {
-    _simpleConsumerFactory = simpleConsumerFactory;
-    _clientId = clientId;
-    _connectTimeoutMillis = connectTimeoutMillis;
-    _metadataOnlyConsumer = true;
-    _simpleConsumer = null;
-
-    // Topic and partition are ignored for metadata-only consumers
-    _topic = null;
-    _partition = Integer.MIN_VALUE;
-
-    initializeBootstrapNodeList(bootstrapNodes);
-    setCurrentState(new ConnectingToBootstrapNode());
-  }
-
-  private SimpleConsumerWrapper(KafkaSimpleConsumerFactory simpleConsumerFactory, String bootstrapNodes,
-      String clientId, String topic, int partition, long connectTimeoutMillis) {
-    _simpleConsumerFactory = simpleConsumerFactory;
+  public SimpleConsumerWrapper(String bootstrapNodes, String clientId, String topic, int partition,
+      long connectTimeoutMillis) {
     _clientId = clientId;
     _topic = topic;
     _partition = partition;
+    if (_topic == null && _partition == Integer.MIN_VALUE) {
```

Comment:
I don't understand this. If topic is null and partition is MIN_VALUE, then what can the code above do with the consumer? Both controller and server need to (and will) specify a partition and topic, right?

Reply:
Topic and partition are ignored for metadata-only consumers. I duplicated the logic we had before for the metadata only consumer